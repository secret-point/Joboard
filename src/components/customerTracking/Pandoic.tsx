import React from "react";
import { Helmet } from "react-helmet";

// Added 2 new customerTracking places unique to the new URLs to track site traffic and completed applications
export const PandoicTracking = () => {

  return (
    <Helmet>
      <script src="https://tracking.pandoiq.com/8950" async />
    </Helmet>
  );
};

export default PandoicTracking;
