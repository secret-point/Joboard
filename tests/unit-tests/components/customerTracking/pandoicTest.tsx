import React from "react";
import Pandoic from "../../../../src/components/customerTracking/Pandoic";
import { mountWithStencil } from "@amzn/stencil-react-components/tests";
import { mount } from "enzyme";

const getItemSpy = jest.spyOn(window.localStorage, "getItem");

describe("Pandoic", () => {

  beforeEach(() => {
    getItemSpy.mockReset();
  });

  it("should match snapshot on bb", () => {
    const url = "https://beta-us.devo.jobsatamazon.hvh.a2z.com/application/";
    Object.defineProperty(window, "location", {
      value: new URL(url)
    } );
    getItemSpy.mockReturnValue("nhe");
    const wrapper = mount(<Pandoic />);
    expect(wrapper).toMatchSnapshot();
  });
});
