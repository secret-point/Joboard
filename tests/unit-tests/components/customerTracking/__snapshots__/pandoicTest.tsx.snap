// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pandoic should match snapshot on bb 1`] = `
<PandoicTracking>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <SideEffect(NullComponent)
      defer={true}
      encodeSpecialCharacters={true}
      script={
        Array [
          Object {
            "async": true,
            "src": "https://tracking.pandoiq.com/8950",
          },
        ]
      }
    >
      <NullComponent
        defer={true}
        encodeSpecialCharacters={true}
        script={
          Array [
            Object {
              "async": true,
              "src": "https://tracking.pandoiq.com/8950",
            },
          ]
        }
      />
    </SideEffect(NullComponent)>
  </HelmetWrapper>
</PandoicTracking>
`;
